### PR TITLE
[EASY] Iterator for OnchainOrderPlacementError

### PR DESCRIPTION
**Pull request by sunce86**
# Description
Inspired by error https://github.com/cowprotocol/services/pull/2574, now the unit test will fail if we forget to add the new variant to database.

## How to test
Existing tests

### CODE REVIEW

File: crates/database/Cargo.toml [hunk 1]
```diff
@@ -12,6 +12,7 @@ const_format = "0.2"
 futures = { workspace = true }
 hex = { workspace = true }
 sqlx = { workspace = true }
+strum = { version = "0.25", features = ["derive"] }
```
Comment (MartinquaXD) [inline crates/database/Cargo.toml:15]: We also have `strum` in the workspace `cargo.toml`.

**Review by MartinquaXD — APPROVED** [review]
Neat 👍

File: crates/database/Cargo.toml (hunk 1 above)
Comment (sunce86) [reply crates/database/Cargo.toml]: fixed

**Review by squadgazzz — APPROVED** [review]

**Review by fleupold — APPROVED** [review]
noice

**Pull request merged**